holopy/inference/tests/test_prior.py

```diff
@@ -17,60 +17,320 @@
 # along with HoloPy.  If not, see <http://www.gnu.org/licenses/>.
 
 
-
+import unittest
 
 from numpy.testing import assert_raises, assert_equal, assert_allclose
 import numpy as np
+from nose.plugins.attrib import attr
 
-from holopy.inference.prior import Gaussian, Uniform
-from holopy.inference import prior
+from holopy.inference.prior import (Prior, Gaussian, Uniform, BoundedGaussian,
+    ComplexPrior, make_center_priors, updated, generate_guess)
 from holopy.inference.result import UncertainValue
+from holopy.core.metadata import data_grid
 from holopy.core.tests.common import assert_obj_close
 from holopy.scattering.errors import ParameterSpecificationError
-#GOLD:log(sqrt(0.5/pi))-1/2
-gold_sigma=-1.4189385332
-
-def test_uniform():
-    u = Uniform(1, 2)
-    assert_equal(u.lnprob(1.4), 0)
-    assert_equal(u.lnprob(0.9), -np.inf)
-    assert_equal(u.lnprob(4), -np.inf)
-    assert_equal(u.guess, 1.5)
-    assert_equal(u.interval, 1)
-    assert_raises(ParameterSpecificationError, Uniform, 4, 6, 7)
-    assert_raises(ParameterSpecificationError, Uniform, 4, 4)
-
-def test_improper_guess():
-    improper1 = Uniform(-2, np.inf)
-    assert_equal(improper1.guess, -2)
-    improper2 = Uniform(-2, np.inf, -1)
-    assert_equal(improper2.guess, -1)
-    improper3 = Uniform(-np.inf, -2)
-    assert_equal(improper3.guess,-2)
-    improper4 = Uniform(-np.inf, np.inf)
-    assert_equal(improper4.guess, 0)
-
-def test_bounded_gaussian():
-    g = prior.BoundedGaussian(1, 1, 0, 2)
-    assert_equal(g.lnprob(-1), -np.inf)
-    assert_equal(g.lnprob(3), -np.inf)
-    assert_equal(g.guess, 1)
-
-def test_gaussian():
-    g = Gaussian(1, 1)
-    assert_equal(g.guess, 1)
-    assert_obj_close(g.lnprob(0),gold_sigma)
-
-def test_complex_prior():
-    p1 = prior.ComplexPrior(Uniform(0,2), Gaussian(2,1))
-    assert_obj_close(p1.real, Uniform(0,2))
-    assert_obj_close(p1.imag, Gaussian(2,1))
-    assert_equal(p1.guess, 1+2j)
-    assert_obj_close(p1.lnprob(0.5+1j), gold_sigma - np.log(2))
-    p2 = prior.ComplexPrior(1, Gaussian(2,1))
-    assert_equal(p2.guess, 1+2j)
-    p3 = prior.ComplexPrior(Uniform(0,2), 2)
-    assert_equal(p3.guess, 1+2j)
+
+GOLD_SIGMA = -1.4189385332 #log(sqrt(0.5/pi))-1/2
+
+class TestBasics(unittest.TestCase):
+    @attr("fast")
+    def test_cannot_instantiate_baseclass(self):
+        self.assertRaises(NotImplementedError, Prior)
+
+class TestUniform(unittest.TestCase):
+    @attr("fast")
+    def test_construction(self):
+        parameters = {'lower_bound':1, 'upper_bound':3, 'guess':2, 'name':'a'}
```

Comment:
pep8!!!!

Reply:
In general it would be good to run these files through a pep8 checker. I like [pep8online.com](www.pep8online.com). It's pretty good overall, but I'm noticing a few mistakes (no space after colons and commas in a few places, for example).